mobile/src/modules/home/components/HomeStackedAreaChart.tsx

```diff
@@ -0,0 +1,159 @@
+import { NavigationInjectedProps } from 'react-navigation';
+import { Button } from '../../../core/components/base/Button';
+import { Colors } from '../../../core/theme/colors';
+import { getBalanceHistoryFromTransactions } from '../../../core/utils/balance/getBalanceHistoryFromTransactions';
+import { BalanceHistoryItem } from '../../../core/utils/balance/typings';
+import { updateAccountTransactions } from '../../auth/store/actions';
+import { Metric, PriceInfo, PriceInfoReduxState } from '../../price-api/store/reducer';
+
+interface Props {
+  accounts: Account[];
+  priceApi: PriceInfoReduxState;
+}
+type TProps = Props;
+
+interface State {
+  priceInBTC: boolean
+}
+
+export class HomeStackedAreaChart extends React.PureComponent<TProps, State> {
+
+  state = {
+    selectedDateRange: 100,
+    priceInBTC: false
+  };
+
+  componentWillMount () {
+    // this.props.dispatch(updateAccountTransactionForMultipleAccounts(this.props.accounts));
+  }
+
+  calculateChartData () {
+    console.log(this.props.priceApi.historicalPriceInfo);
+    const accountTransactions = this.props.accounts.map((account) => {
+      if (account.transactions) {
+        return getBalanceHistoryFromTransactions(account.account,
+          convertNQTStringToNumber(account.balanceNQT), account.transactions);
+      }
+    });
+    const now = new Date();
+    const data = [];
+    const historicalPrice: Metric[] = [];
+    let i = 0;
+    for (const d = new Date(new Date().setDate(now.getDate() - this.state.selectedDateRange));
+      d <= now; d.setDate(d.getDate() + 1)) {
+      data[i] = {
+        day: new Date(d)
+      };
+
+      if (this.state.priceInBTC) {
+        const price = this.props.priceApi.historicalPriceInfo.Data.find((metric) => {
+          return this.sameDay(new Date(metric.time * 1000), d);
+        });
+        historicalPrice[d.getTime()] = price || this.props.priceApi.historicalPriceInfo.Data[0];
+      }
+      accountTransactions.map((transactions, accountIndex) => {
+        // set to 0 by default, in case we can't match a transaction
+        data[i][`account${accountIndex}`] = 0;
+
+        if (transactions && transactions.length) {
+          for (let t = transactions.length - 1; t >= 0; t--) {
+            const transaction = transactions[t];
+            // only match up to the most recent transaction, not beyond
+            if (convertBurstTimeToDate(transaction.timestamp) > d) {
+              continue;
+            }
+            if (this.state.priceInBTC) {
+              const priceInBTCOnThatDay = historicalPrice[d.getTime()];
+              data[i][`account${accountIndex}`] = transaction.balance * priceInBTCOnThatDay.close;
+            } else {
+              data[i][`account${accountIndex}`] = transaction.balance;
+            }
+          }
+        }
+      });
+      i++;
+    }
+
+    return data;
+  }
+
+  toggleBTCPrice = () => {
+    this.setState({ priceInBTC: !this.state.priceInBTC });
+  }
+
+  sameDay (d1, d2) {
```

Comment:
looks like `export const isSameDay = (d1, d2) => { ... } ` in `core/utils/date.ts`

Reply:
and types, yea

Reply:
nice find!